googleprediction/gp-connector/gp-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/googleprediction-config/getAccessTokenByRefreshToken.xml

```diff
@@ -0,0 +1,82 @@
+    <parameter name="clientSecret"
+               description="clientSecret of your App, given by google console when you registered your application"/>
+    <parameter name="refreshToken" description="Refresh Token"/>
+    <sequence>
+        <property name="uri.var.refreshToken" expression="$func:refreshToken"/>
+        <property name="uri.var.clientId" expression="$func:clientId"/>
+        <property name="uri.var.clientSecret" expression="$func:clientSecret"/>
+        <payloadFactory media-type="json">
+            <format>
+                {
+                "refresh_token": "$1",
+                "grant_type": "refresh_token",
+                "client_secret": "$2",
+                "client_id": "$3"
+                }
+            </format>
+            <args>
+                <arg expression="$ctx:uri.var.refreshToken"/>
+                <arg expression="$ctx:uri.var.clientSecret"/>
+                <arg expression="$ctx:uri.var.clientId"/>
+            </args>
+        </payloadFactory>
+        <property name="messageType" value="application/x-www-form-urlencoded" scope="axis2"/>
+        <!-- Remove custom Headers from the API Response -->
+        <header name="Content-Encoding" action="remove" scope="transport"/>
+        <property name="REQUEST_HOST_HEADER" scope="axis2" value="www.googleapis.com"/>
+        <header name="Transfer-Encoding" action="remove" scope="transport"/>
+        <call>
+            <endpoint>
+                <http method="post"
+                      uri-template="https://www.googleapis.com/oauth2/v3/token?grant_type=refresh_token&amp;client_id={+uri.var.clientId}&amp;client_secret={+uri.var.clientSecret}&amp;refresh_token={+uri.var.refreshToken}"/>
+            </endpoint>
+        </call>
+        <!-- Setting the new access token to the property. -->
+        <property name="uri.var.accessToken" expression="json-eval($.access_token)"/>
+        <script language="js">
+            <![CDATA[
+            importPackage(Packages.org.apache.synapse.config);
+            var newAccessToken = mc.getProperty('uri.var.accessToken');
+            var newAccessRegistryPath = mc.getProperty('uri.var.accessTokenRegistryPath');
+                if(newAccessToken != null && newAccessToken != ''){
+                var existingAccessToken =mc.getConfiguration().getRegistry().getResource(newAccessRegistryPath);
+                    if(existingAccessToken == null){
+                        // Create the registry entry if no such entry exists.
+                        mc.getConfiguration().getRegistry().newResource(newAccessRegistryPath,false);
+                        mc.getConfiguration().getRegistry().updateResource(newAccessRegistryPath,newAccessToken);
+                    } else {
+                        // Update the registry entry if it already exists.
+                        mc.getConfiguration().getRegistry().updateResource(newAccessRegistryPath,newAccessToken);
```

Comment:
Since updating the access token is written in both if and else, can we move this outside of the condition?